internal/expr/prepare.go

```diff
@@ -211,11 +217,11 @@ func (pe *ParsedExpr) Prepare(args ...any) (expr *PreparedExpr, err error) {
 	// Generate and save reflection info.
 	for _, arg := range args {
 		if arg == nil {
-			return nil, fmt.Errorf("need struct or map, got nil")
+			return nil, fmt.Errorf("need valid value, got nil")
 		}
 		t := reflect.TypeOf(arg)
 		switch t.Kind() {
-		case reflect.Struct, reflect.Map:
+		case reflect.Struct, reflect.Map, reflect.Slice, reflect.Array:
```

Comment:
If we are planning to remove array in the future for the range syntax, should we do it know to later preserve backwards compatibility?

Reply:
Removed array support. If it is not going to work with the new slice of slice syntax we are going to introduce then its not worth having here. Also if you do have an array, it could be passed to the query by doing `s := append(sqlair.S{}, arr...)` and using `S` in the query.